Source/Scene/Cesium3DTileset.js

```diff
@@ -1755,14 +1911,110 @@ define([
         var selectedTiles = tileset._selectedTiles;
         var length = selectedTiles.length;
         var tileVisible = tileset.tileVisible;
-        for (var i = 0; i < length; ++i) {
-            var tile = selectedTiles[i];
-            if (tile.selected) {
-                // Raise visible event before update in case the visible event
-                // makes changes that update needs to apply to WebGL resources
-                tileVisible.raiseEvent(tile);
-                tile.update(tileset, frameState);
-                incrementPointAndFeatureSelectionCounts(tileset, tile.content);
+
+        var tile, i;
+
+        if (tileset._refining && frameState.context.stencilBuffer && length > 0) {
+            /**
+             * Consider 'leaf' tiles as selected tiles that have no selected descendants. They may have children,
+             * but they are currently our effective leaves because they do not have selected descendants. These tiles
+             * are those where with tile._finalResolution === true.
+             * Let 'unresolved' tiles be those with tile._finalResolution === false.
+             *
+             * 1. Render just the backfaces of unresolved tiles in order to lay down z
+             * 2. Render all frontfaces whereever tile._selectionDepth > stencilBuffer.
+             *    Replace stencilBuffer with tile._selectionDepth, when passing the z test.
+             *    Because children are always drawn before ancestors (@see {@link traverseAndSelect}),
+             *    this effectively draws children first and does not draw ancestors if a descendant has already
+             *    been drawn at that pixel.
+             *    Step 1 prevents child tiles from appearing on top when they are truly behind ancestor content.
+             *    If they are behind the backfaces of the ancestor, then they will not be drawn.
+             *
+             * NOTE: Step 2 sometimes causes visual artifacts when backfacing child content has some faces that
+             * partially face the camera and are inside of the ancestor content. Because they are inside, they will
+             * not be culled by the depth writes in Step 1, and because they partially face the camera, the stencil tests
+             * will draw them on top of the ancestor content.
+             *
+             * NOTE: Because we always render backfaces of unresolved tiles, if the camera is looking at the backfaces
+             * of an object, they will always be drawn while loading, even if backface culling is enabled.
+             */
+
+            commandList.push(stencilClearCommnad);
```

Comment:
Okay, I think the multiple tilesets thing will be fine because we redistribute them into frustums but it doesn't change ordering for things that go into the same frustum, like a stable sort. So for a particular frustum, a tileset's commands should all be contiguous.

But the thing we need to do is clear stencil before each tileset for each frustum. Is there a way we can accomplish this?

ex.)
Frustum 1: Clear stencil, render tileset 1, clear stencil, render tileset 2
Frustum 2: Clear stencil, render tileset 1, clear stencil, render tileset 2